### PR TITLE
[REF] web_editor,web: reuse the autoscroll on drag feature

### PR DESCRIPTION
**Pull request by IT-Ideas**
Prior to this commit, the autoscroll on drag feature was accessible and only
used in the web_editor application.

After this commit, the autoscroll on drag feature will be available in the
web application and has been refactored in order to allow its usage
in other application (as for the sign application).

task-2280965

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/js/widgets/smooth_scroll.js [hunk 1]
```diff
@@ -0,0 +1,315 @@
+odoo.define('web.smooth_scroll_on_drag', function (require) {
```
Comment (qsm-odoo) [inline addons/web/static/src/js/widgets/smooth_scroll.js:1]: This returns a class so we normally give the name of that class, so SmoothScrollOnDrag. Also the file was placed inside the widgets folder but this is not a widget... but I don't know... @aab-odoo could you tell us where to put this? :)

File: addons/web/static/src/js/widgets/smooth_scroll.js [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+        this.autoScrollHandler = null;
+        this.scrollDecelerator = {
+            vertical: 0,
+            horizontal: 0,
+        };
+
+        this.scrollStepDirection = {
+            vertical: this.scrollStepDirectionEnum.down,
+            horizontal: this.scrollStepDirectionEnum.right,
+        }
```
Comment (qsm-odoo) [inline addons/web/static/src/js/widgets/smooth_scroll.js:117]: Missing `;` here, check that your linter is still working

File: addons/web/static/src/js/widgets/smooth_scroll.js [hunk 3]
```diff
@@ -0,0 +1,315 @@
+odoo.define('web.smooth_scroll_on_drag', function (require) {
+"use strict";
+
+const Class = require('web.Class');
+const Mixins = require('web.mixins');
```
Comment (qsm-odoo) [inline addons/web/static/src/js/widgets/smooth_scroll.js:5]: No uppercase letter for `mixins`

File: addons/web/static/src/js/widgets/smooth_scroll.js [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+
+/**
+ * Provides a helper for SmoothScrollOnDrag options.offsetElements
+ */
+const OffsetElementsHelper = Class.extend({
```
Comment (qsm-odoo) [inline addons/web/static/src/js/widgets/smooth_scroll.js:10]: As discussed in discord, I think there is a problem with this. I don't know about the logic but while testing there are new problems that are not there in master.

File: addons/web/static/src/js/widgets/smooth_scroll.js (hunk 1 above)
Comment (aab-odoo) [reply addons/web/static/src/js/widgets/smooth_scroll.js:1]: Well the convention about JS module names changed recently (I expected @seb-odoo to communicate about it). It is now the path of the file, starting from module name: https://github.com/odoo/odoo/blob/master/addons/mail/static/src/widgets/discuss/discuss.js#L1

Comment (aab-odoo) [reply addons/web/static/src/js/widgets/smooth_scroll.js:1]: For the initial question, I'd say in /core, even though I'm not convinced. But according to what it contains, and the other folders, it seems to be the appropriate place.

Comment (qsm-odoo) [reply addons/web/static/src/js/widgets/smooth_scroll.js:1]: JKE told me it was planned but I was indeed waiting for a communication about it. @IT-Ideas You can do as @aab-odoo says, sorry :wink:

Comment (seb-odoo) [reply addons/web/static/src/js/widgets/smooth_scroll.js]: I didn't really get time to work on the doc yet. I just now created task-2293648 for this, we probably want to come with the final version together.

**Comment by tivisse** [comment]
@robodoo r+

**Comment by qsm-odoo** [comment]
FYI: https://github.com/odoo/odoo/pull/54311

**Pull request closed without merging**